component/site/libraries/dbmodel.php

```diff
@@ -2580,7 +2580,20 @@ function listIcalEventsByRange($startdate, $enddate, $limitstart, $limit, $showr
 		{
 			$query .="\n GROUP BY rpt.rp_id";
 		}
-
+/*
+		//Dirty hack to include locations categories grouping
+		if(strpos($query, "loccatmap"))
+		{
+			if(strpos($query, 'GROUP BY'))
+			{
+				$query .= ", loc.loc_id";
+			}
+			else
+			{
+				$query .= "GROUP BY loc.loc_id";
+			}
+		}
+*/
```

Comment:
Geraint, I had to apply this to a user site to make it work. Nevertheless after all this time with all the updates he has not complain so if you do not feel confortable with this change you can reject it. After all this was only for older MySQL versions.